src/term/term_to_net.rs

```diff
@@ -159,15 +169,21 @@ fn encode_term(
     // - 1: points to the occurrence of the first variable.
     // - 2: points to the occurrence of the second variable.
     // core: & val ~ {lab fst snd} (val not necessarily main port)
-    Term::Dup { fst, snd, val, nxt } => {
-      let dup = inet.new_node(Dup { lab: *dups });
-      *dups += 1;
-      let val = encode_term(inet, val, Port(dup, 0), scope, vars, global_vars, dups);
+    Term::Dup { fst, snd, val, nxt, tag } => {
+      let dup: u32;
+      if let Some(tag) = tag {
```

Comment:
I think it would be better to have this part only return the label.
If explicitly tagged and new tag, give a new label.
If explicitly tagged and repeated tag, use the label we decided on before.
If implicitly tagged, use the implicit label counter.

Then we create the dup node and insert it into the inet